### ISSUE TITLE
Playing Spotify ID items via play_uri() results in UPnP 701

### ISSUE THREAD

**Issue by fluxens** [SoCo/SoCo#762]
From what I understand this should work:

`s = SoCo("192.168.4.113")`
`s.play_uri("x-sonos-spotify:spotify:album:6xaalWgqmC00dKgrFLlcmk")`

This results in  `UPnP Error 701 received: Transition not available from 192.168.4.113`.

 I guess I am missing smth.

**Comment by pwt** [comment]
There are issues with playback from music services, which are being looked at. See the 'WARNING' section in the README (https://github.com/SoCo/SoCo/blob/master/README.rst), and issue #557.

A workaround is first to add the album as a Sonos Playlist, then use SoCo to add it to the queue, followed by playing from the queue.

**Comment by amelchio** [comment]
I have written a custom `add_spotify_uri_to_queue()` that may be able to help you: https://github.com/amelchio/pysonos/blob/v0.0.37/pysonos/core.py#L1591-L1691

```
>>> import pysonos
>>> a=pysonos.SoCo("10.0.2.14")
>>> a.add_spotify_uri_to_queue("spotify:album:6xaalWgqmC00dKgrFLlcmk")
1
>>> print(a.get_queue()[0].album)
Wildflower
```

**Comment by acatoire** [comment]
Hi, I am also interested in a simple way to queue knew spotify playlist/album/track by id
@amelchio your code was in 0.37 but is not present anymore in 0.51 has it been mooved?
Is there a new simple official way to play spotify elements?

My project actually use https://github.com/jishi/node-sonos-http-api where it is possible (without auth) to play any  spotify playlist/album/track with from its id. I would like to use soco instead to have everything in python.

**Comment by amelchio** [comment]
Yeah, I have moved the code to a plugin and actually submitted it to SoCo in #838. I hope it will be in SoCo once v0.23 is released.

You should be able to reuse the plugin file from the PR in your own project with very few changes if you are in a hurry.

**Comment by pwt** [comment]
Fixed by #838 in v0.23.
